### PR TITLE
PTypes 27: Convert Region.visit to use PType

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: hail/src/test/scala/is/hail/annotations/StagedRegionValueSuite.scala [hunk 1]
```diff
@@ -117,21 +117,21 @@ class StagedRegionValueSuite extends SparkSuite {
...
-    assert(rv.pretty(rt.virtualType) == rv2.pretty(rt.virtualType))
+    assert(rv.pretty(rt) == rv2.pretty(rt))
     assert(rv.region.loadInt(rt.loadElement(rv.region, rv.offset, 0)) ==
       rv2.region.loadInt(rt.loadElement(rv2.region, rv2.offset, 0)))
   }
 
   @Test
   def testStruct() {
-    val rt = TStruct("a" -> TString(), "b" -> TInt32())
+    val rt = TStruct("a" -> TString(), "b" -> TInt32()).physicalType
```
Comment (cseed) [inline hail/src/test/scala/is/hail/annotations/StagedRegionValueSuite.scala]: Do we prefer `PStruct("a" => ...)`?

Comment (tpoterba) [reply hail/src/test/scala/is/hail/annotations/StagedRegionValueSuite.scala]: I'm not sure. This stuff is going to change at some point (in the PType 50s?)  `.physicalType` is on the chopping block, and so using it is a marker to think about it later.

I'm thinking about something like `Canonical(t)` as a way to construct the canonical physical type from a virtual type (which is also constructable).

Comment (tpoterba) [reply hail/src/test/scala/is/hail/annotations/StagedRegionValueSuite.scala]: I don't totally love _constructing_ physical types all over the code. I think there should be fewer physical entry points.

Comment (cseed) [reply hail/src/test/scala/is/hail/annotations/StagedRegionValueSuite.scala]: OK, great, I think `Canonical(t)` is clearer in the long-term.  I mostly wanted to raise the question, I'll approve for now.

**Pull request merged**